DataSetExplorer/Controllers/Annotations/AnnotationController.cs

```diff
@@ -83,36 +82,26 @@ public IActionResult UpdateAnnotation([FromRoute] int id, [FromBody] AnnotationD
                 return BadRequest(new { message = e.Message });
             }
         }
-
+        
         [HttpGet]
-        [Route("requiring-additional-annotation")]
-        public IActionResult FindInstancesRequiringAdditionalAnnotation([FromQuery(Name = "projectIds")] string projectIds)
+        [Route("requiring-additional-annotation/{id}")]
+        public IActionResult FindInstancesRequiringAdditionalAnnotation([FromRoute] int id)
         {
-            return FindInstances(projectIds, _dataSetAnalysisService.FindInstancesRequiringAdditionalAnnotation);
+            return FindInstances(id, _dataSetAnalysisService.FindInstancesRequiringAdditionalAnnotation);
         }
 
         [HttpGet]
-        [Route("disagreeing-annotations")]
-        public IActionResult FindInstancesWithAllDisagreeingAnnotations([FromQuery(Name = "projectIds")] string projectIds)
+        [Route("disagreeing-annotations/{id}")]
+        public IActionResult FindInstancesWithAllDisagreeingAnnotations([FromRoute] int id)
         {
-            return FindInstances(projectIds, _dataSetAnalysisService.FindInstancesWithAllDisagreeingAnnotations);
+            return FindInstances(id, _dataSetAnalysisService.FindInstancesWithAllDisagreeingAnnotations);
         }
 
-        private IActionResult FindInstances(string projectIds, Func<IEnumerable<int>, Result<List<SmellCandidateInstances>>> findInstancesMethod)
+        private IActionResult FindInstances(int projectId, Func<int, Result<List<SmellCandidateInstances>>> findInstancesMethod)
```

Comment:
"findInstancesMethod" is a lazy name - it is redundant (the first two words repeat the name of the method, while the third word repeats the name of the param's type). Maybe searchCriteria or something more descriptive?